framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -896,31 +910,63 @@ protected String dotOutputFileName(UnderlyingAST ast) {
         return null;
     }
 
+    /**
+     * Returns the type of the left-hand side of an assignment without
+     * applying local variable defaults to type variables.
+     *
+     * The type variables that are types of local variables are defaulted to
```

Comment:
I have to admit that I find the behavior for type variables rather confusing. "@Nullable T" is not a supertype of "T", so defaulting local type variables to the top type would prevent simple code like:

```
<T> T foo(T p) {
  T local = p;
  return local;
}
```

So isn't there more going on than just using the top type for local variables? If we are doing the same thing for type variables and reference types, why would type variables need special treatment?